### ISSUE TITLE
How to Configure Operational Data in yang modules?

### ISSUE THREAD

**Issue by aswin19996** [CESNET/netopeer2#748]
How to instantiate values for a read only leaf in yang modules?
For Example:

> container operational-state {
>       config false;
>       description
>         "Operational state for the O-RAN network element";
>       leaf restart-cause {
>         type enumeration {
>           enum POWER-ON {
>             description
>               "Equipment restarted because it was powered on";
>           }
>           enum SUPERVISION-WATCHDOG {
>             description
>               "Equipment restarted because it's supervision wathcdog timer wasn't reset
>               by a NETCONF client (inferring loss of NETCONF connectivity)";
>           }
>           enum MPLANE-TRIGGERED-RESTART {
>             description
>               "Equipment restarted because of an M-plane issued  rpc";
>           }
>           enum SOFTWARE-FAILURE {
>             description
>               "Equipment restarted because of software failure";
>           }
>           enum OTHER-WATCHDOG-TIMER {
>             description
>               "Equipment restarted because of some other non NETCONF watchdog timer";
>           }
>           enum UNKNOWN {
>             description
>               "The restart reason for the Equipment is unknown";
>           }
>         }
>         description "the cause for the last restart of the O-RAN Network Element";
>       }

How does the leaf restart-cause in this container gets a value instantiated?

**Comment by michalvasko** [comment]
It should all be described in the [docs](https://netopeer.liberouter.org/doc/sysrepo/master/html/index.html#oper_data).

**Comment by robbynet** [comment]
Also working on O-RAN, so let me provided a detailed answer : 
Application attached to sysrepo needs to : 
1) Subscribes for providing the operational data  => sr_oper_get_items_subscribe()
2) Implement the callback shared in 1)
Typical example is provided [here](https://github.com/sysrepo/sysrepo/blob/v1.4.70/examples/oper_data_example.c)

**Comment by aswin19996** [comment]
Thanks @michalvasko. 
Also Thanks @robbynet , and i would like to know how this can be implemented. Is it that i will have to write a C code, that will have a function call to  sr_oper_get_items_subscribe(), to subscribe for the operational data?
or is it implemented in any other way?

**Comment by michalvasko** [comment]
Yes, you have to write some C code, specifically an implementation of an operational callback that will populate your data and subscribe it with `sr_oper_get_items_subscribe()`. Or you can use either of the [bindings](https://github.com/sysrepo/sysrepo/#useful-cmake-sysrepo-options).
